### ISSUE TITLE
selecting no variables  in recipes selections in v0.1.15

### ISSUE THREAD

**Issue by kylegilde** [tidymodels/recipes#603]

**Comment by kylegilde** [comment]
Hi, 

I was checking out the fully supported tidyselect functionality, and I was wondering if the behavior changed when a logical expression selects no columns. 

Before this version, it would error out when prepped, but in 0.1.15, it appears that it now will prep and with no changes occurring in the step if no columns are selected. Is this correct?

Thank you!

**Comment by topepo** [comment]
Can you give an example?

**Comment by kylegilde** [comment]
I think this produced an error in earlier recipes versions, and now it seems to handle the no-selection case pretty well now.

> library(recipes)
> rec <- 
+   recipe(as.formula("~."), data = mtcars) %>%
+   step_dummy(all_nominal()) %>% 
+   prep()
> rec
Data Recipe

Inputs:

      role #variables
 predictor         11

Training data contained 32 data points and no missing data.

Operations:

Dummy variables were *not* created since no columns were selected. [trained]

**Comment by topepo** [comment]
So I'm pretty sure that it depends on the step. We have _not_ made a global change so that any step is skipped if nothing is selected.

**Comment by DavisVaughan** [comment]
That step in particular always worked fine with empty selections

New recipes (0.1.15)

``` r
library(recipes)
rec <- recipe(mpg ~ ., mtcars)
rec <- step_dummy(rec, starts_with("foobar"))
prep(rec)
#> Data Recipe
#> 
#> Inputs:
#> 
#>       role #variables
#>    outcome          1
#>  predictor         10
#> 
#> Training data contained 32 data points and no missing data.
#> 
#> Operations:
#> 
#> Dummy variables were *not* created since no columns were selected. [trained]
```

Old recipes (0.1.14)

``` r
library(recipes)
rec <- recipe(mpg ~ ., mtcars)
rec <- step_dummy(rec, all_nominal())
prep(rec)
#> Data Recipe
#> 
#> Inputs:
#> 
#>       role #variables
#>    outcome          1
#>  predictor         10
#> 
#> Training data contained 32 data points and no missing data.
#> 
#> Operations:
#> 
#> Dummy variables were *not* created since no columns were selected. [trained]
```

---

However, it seems I may have inadvertently changed this behavior with other steps when I did the tidyselect updates.

New recipes (0.1.15)

``` r
library(recipes)
rec <- recipe(mpg ~ ., mtcars)
rec <- step_center(rec, starts_with("foobar"))
prep(rec)
#> Data Recipe
#> 
#> Inputs:
#> 
#>       role #variables
#>    outcome          1
#>  predictor         10
#> 
#> Training data contained 32 data points and no missing data.
#> 
#> Operations:
#> 
#> Centering for <none> [trained]
```

Old recipes (0.1.14)

``` r
library(recipes)
rec <- recipe(mpg ~ ., mtcars)
rec <- step_center(rec, starts_with("foobar"))
prep(rec)
#> Error: No variables or terms were selected.
```

<sup>Created on 2020-11-13 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

**Comment by kylegilde** [comment]
@DavisVaughan , I'm glad to know that it wasn't just my imagination! 

I would say that I **strongly prefer** the **new behavior** where an error does NOT occur. It makes the recipe more dynamic & reusable on different datasets. 

It would helpful if the selections doc makes note of this behavior.

https://github.com/tidymodels/recipes/blob/master/R/selections.R

Also, this line in the doc is out of date:

Additionally, the tidyselect domain specific
#' language is not recognized here, meaning that `&`, `|`, `!`, and `-`
#' will not work.

**Comment by DavisVaughan** [comment]
> Also, this line in the doc is out of date:

That is specifically about `step_interact()`, and should still hold

**Comment by juliasilge** [comment]
Bigger picture, related to #531

**Comment by juliasilge** [comment]
Closed in #813 🎉
